src/sprites/AbilityPanel.lua

```diff
@@ -47,103 +47,56 @@ function AbilityPanel.getInstance() return _instance end
 
-  self:moveTo(0, panelShownY)
+  self:setCenter(0, 0)
+  self:moveTo(0, 0)
+  self:setZIndex(99)
   self:add()
-
-  spriteOne:moveTo(16, buttonShownY)
-  spriteOne:setZIndex(100)
-  spriteOne:add()
-  spriteTwo:moveTo(48, buttonShownY)
-  spriteTwo:setZIndex(100)
-  spriteTwo:add()
-  spriteThree:moveTo(80, buttonShownY)
-  spriteThree:setZIndex(100)
-  spriteThree:add()
-
   self:setIgnoresDrawOffset(true)
-  spriteOne:setIgnoresDrawOffset(true)
-  spriteTwo:setIgnoresDrawOffset(true)
-  spriteThree:setIgnoresDrawOffset(true)
-
-  self.abilitiesCount = 1
-  self:setZIndex(99)
-end
-
-local _spriteAdd = AbilityPanel.add
-function AbilityPanel:add()
-  _spriteAdd(self)
 
-  spriteOne:add()
-  spriteTwo:add()
-  spriteThree:add()
-end
-
-function AbilityPanel:shake(shakeTime, shakeMagnitude)
-  local shakeTimer = pd.timer.new(shakeTime, shakeMagnitude, 0)
-
-  shakeTimer.updateCallback = function(timer)
-    local magnitude = math.floor(timer.value)
-    local shakeX = math.random(-magnitude, magnitude)
-    local shakeY = math.random(-magnitude, magnitude)
-    self:moveTo(self.original_x + shakeX, self.original_y + shakeY)
+  for i, sprite in ipairs(buttonSprites) do
+    sprite:moveTo(spritePositions[i]:unpack())
+    sprite:setZIndex(100)
+    sprite:add()
+    sprite:setIgnoresDrawOffset(true)
   end
 
-  shakeTimer.timerEndedCallback = function()
-    self:moveTo(self.original_x, self.original_y)
-  end
+  self:setUpdatesEnabled(false)
 end
 
-function AbilityPanel:addItem(item)
-  if self.abilitiesCount == 3 then
-    table.remove(items, 1)
-    table.insert(items, item)
+-- gfx.sprite overrides to manage child button sprites along with sprite.
 
-    self:updateItemImages()
-  else
-    table.insert(items, item)
+function AbilityPanel:add()
+  AbilityPanel.super.add(self)
 
-    self:updateItemsCount()
-    self:updateItemImages()
+  for _, sprite in ipairs(buttonSprites) do
+    sprite:add()
   end
 end
 
-function AbilityPanel:cleanUp()
-  self:setItems()
-
-  self:remove()
-end
+function AbilityPanel:remove()
+  AbilityPanel.super.remove(self)
 
-function AbilityPanel:removeRightMost()
-  if self.abilitiesCount == 1 then
-    items[1] = nil
-  elseif self.abilitiesCount == 2 then
-    items[2] = nil
-  elseif self.abilitiesCount == 3 then
-    items[3] = nil
+  for _, sprite in ipairs(buttonSprites) do
+    sprite:remove()
   end
-
-  self:updateItemsCount()
-  self:updateItemImages()
 end
 
-function AbilityPanel:setItems(item1, item2, item3)
-  items[1] = item1
-  items[2] = item2
-  items[3] = item3
+-- Update function - reads player blueprints and updates accordingly.
 
-  self:updateItemsCount()
-  self:updateItemImages()
-end
+function AbilityPanel:updateBlueprints()
+  local blueprints = Player.getInstance().blueprints
+  self.blueprints = blueprints
 
-function AbilityPanel:updateItemsCount()
-  self.abilitiesCount = #items
-end
+  for i, sprite in ipairs(buttonSprites) do
+    if blueprints[i] then
+      sprite:add()
```

Comment:
aren't the sprites already added when AbilityPanel is [here](https://github.com/Franchovy/PlayJam-5/pull/2/files#diff-7f6dc5a823571736a8d4c4e52c74b697c0e6b75cde2bc4edfb5a9d78ce223842R74)?

Reply:
Point valid, I'll make sure to correct that